src/passes/literalExpressionEvaluator/literalExpressionEvaluator.ts

```diff
@@ -128,15 +129,43 @@ function evaluateBinaryLiteral(node: BinaryOperation): RationalLiteral | boolean
   if (left === null || right === null) {
     // In some cases a binary expression could be calculated at
     // compile time, even when only one argument is a literal.
+
+    let rightExpression;
+    if (left === null) {
+      rightExpression = true;
+    } else {
+      rightExpression = false;
+    }
+
     const notNullMember = left ?? right;
     if (notNullMember === null) {
       return null;
     } else if (typeof notNullMember === 'boolean') {
       switch (node.operator) {
         case '&&': // false && x = false
-          return notNullMember ? null : false;
+          if (notNullMember === true) {
```

Comment:
This is a bad programming practice, since you already know that `notNullMember` is a boolean value. Please change it to:
```suggestion
          if (notNullMember) {
```